dspace-api/src/main/java/org/dspace/storage/bitstore/SyncS3BitStoreService.java

```diff
@@ -53,29 +58,42 @@ public void put(Bitstream bitstream, InputStream in) throws IOException {
         String key = getFullKey(bitstream.getInternalId());
         //Copy istream to temp file, and send the file, with some metadata
         File scratchFile = File.createTempFile(bitstream.getInternalId(), "s3bs");
-        try {
-            FileUtils.copyInputStreamToFile(in, scratchFile);
-            long contentLength = scratchFile.length();
-            // The ETag may or may not be and MD5 digest of the object data.
-            // Therefore, we precalculate before uploading
-            String localChecksum = org.dspace.curate.Utils.checksum(scratchFile, CSA);
+        try (
+                FileOutputStream fos = new FileOutputStream(scratchFile);
```

Comment:
is the in.close() after this important?
it is strange to close InputStream here, or?

Reply:
It is from Vanilla I just copied it